src/GraphODataTemplateWriter/CodeHelpers/ObjC/TypeHelperObjC.cs

```diff
@@ -121,9 +125,16 @@ public static string GetTypeString(this OdcmType type)
                 case "Stream":
                     return "NSStream";
                 case "Duration":
-                    return "Duration";
+                    return "MSDate";
```

Comment:
Are you referring to this [MSDate](https://github.com/microsoftgraph/msgraph-sdk-ios/blob/dev/MSGraphSDK/MSGraphCoreSDK/Core/MSDate.m)? I don't think this is correct. This is a [ISO 8601]([url](http://docs.oasis-open.org/odata/odata-csdl-xml/v4.01/cs01/odata-csdl-xml-v4.01-cs01.html#_Toc505863963)) duration, not a date. Either we provide a Duration object, or we handle converting to NSTimeInterval.

Reply:
I have made change related to duration.
Have added a MSDuration class in set of hand written models and have added the code in templates to use methods from this class. 
Please have a look.